pkg/client/mapper.go

```diff
@@ -28,63 +28,78 @@ import (
 	"github.com/abcxyz/team-link/pkg/groupsync"
 )
 
-type GoogleGroupToGitHubMapper struct {
-	GoogleGroupToGitHubTeam map[string][]string
-}
+// GroupMapper implements groupsync.OneToManyGroupMapper
+// For now it was be used as Mapper between GoogleGroup
+// and GitHubTeam.
+// Both GoogleGroupMapper and GitHubMapper have similar
+// variables and functions, this way we don't need to repeatly
+// write similar function for both on them.
+type GroupMapper map[string][]string
 
-// AllGroupIDs returns the set of groupIDs being mapped (the key set).
-func (g *GoogleGroupToGitHubMapper) AllGroupIDs(ctx context.Context) ([]string, error) {
+func (m GroupMapper) AllGroupIDs(ctx context.Context) ([]string, error) {
 	res := make([]string, 0)
-	for key := range g.GoogleGroupToGitHubTeam {
+	for key := range m {
```

Comment:
`slices.Collect` is in go 1.23, this project is still using go 1.22, will try update go version and update this API here later.